### PR TITLE
Adopt the npx style with a new script

### PR DESCRIPTION
**Pull request by mzyil**

### CODE REVIEW

File: check-environment-and-install [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+return_code=0
+if [[ -n "${BROWSER_DRIVER_INSTALLER_CHROME_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH:-}" ]]; then
+    npx browserDriverInstaller --browser-name Chrome                                                \
+                    --browser-version "$BROWSER_DRIVER_INSTALLER_CHROME_VERSION"                    \
+                    --target-path "$BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH" || return_code=$?
+else
+    echo "The environment variables for Chrome are not set, skipping post-install for Chrome"
```
Comment (thehcma) [inline check-environment-and-install]: list the environment variables themselves

Comment (thehcma) [reply check-environment-and-install]: _the_ post-install _step_

File: check-environment-and-install [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+fi
+
+if [[ -n "${BROWSER_DRIVER_INSTALLER_FIREFOX_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH:-}" ]]; then
+    npx browserDriverInstaller --browser-name Firefox                                               \
+                    --browser-version "$BROWSER_DRIVER_INSTALLER_FIREFOX_VERSION"                   \
+                    --target-path "$BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH" || return_code=$?
+else
+    echo "The environment variables for Firefox are not set, skipping post-install for Firefox"
```
Comment (thehcma) [inline check-environment-and-install]: same as above

File: check-environment-and-install [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
```
Comment (thehcma) [inline check-environment-and-install:1]: add the same guard we have in our scripts

also please run this script via ``ucfa``

File: check-environment-and-install [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+return_code=0
```
Comment (thehcma) [inline check-environment-and-install]: return code (a Python concept) and exit status are 2 different things

here it's exit_status

File: check-environment-and-install [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+if [[ -n "${BROWSER_DRIVER_INSTALLER_CHROME_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH:-}" ]]; then
+    npx browserDriverInstaller --browser-name Chrome                                                \
+                    --browser-version "$BROWSER_DRIVER_INSTALLER_CHROME_VERSION"                    \
+                    --target-path "$BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH" || return_code=$?
```
Comment (thehcma) [inline check-environment-and-install]: rather than || - either use an exit trap or (set -e) as we do in our shell scripts

**Review by thehcma — CHANGES_REQUESTED** [review]

File: check-environment-and-install (hunk 1 above)
Comment (mzyil) [reply check-environment-and-install]: listed the env var names, changed the text to: `the installation step`

File: check-environment-and-install (hunk 2 above)
Comment (mzyil) [reply check-environment-and-install]: fixed

File: check-environment-and-install (hunk 3 above)
Comment (mzyil) [reply check-environment-and-install:1]: fixed

File: check-environment-and-install (hunk 4 above)
Comment (mzyil) [reply check-environment-and-install]: fixed

File: check-environment-and-install (hunk 5 above)
Comment (mzyil) [reply check-environment-and-install]: We want the execution to continue and finish the other installation if one of the installations fail, no?

**Comment by mzyil** [comment]
> _the_ installation

fixed

File: check-environment-and-install (hunk 5 above)
Comment (thehcma) [reply check-environment-and-install]: I think not

Comment (mzyil) [reply check-environment-and-install]: Fixed

File: postinstall [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+set -e
 if [[ -n "${BROWSER_DRIVER_INSTALLER_CHROME_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH:-}" ]]; then
     node ./index.js --browser-name Chrome                                        \
                     --browser-version "$BROWSER_DRIVER_INSTALLER_CHROME_VERSION" \
-                    --target-path "$BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH" || return_code=$?
-else 
-    echo "The environment variables for Chrome are not set, skipping post-install for Chrome"
+                    --target-path "$BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH"
+else
+    echo "The environment variables for Chrome (\$BROWSER_DRIVER_INSTALLER_CHROME_VERSION, \$BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH) are not set, skipping the post-install step for Chrome"
```
Comment (thehcma) [inline postinstall]: remove the $ (it's not part of the env var name)

File: postinstall [hunk 7]
```diff
@@ -1,20 +1,20 @@
...
 
 if [[ -n "${BROWSER_DRIVER_INSTALLER_FIREFOX_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH:-}" ]]; then
     node ./index.js --browser-name Firefox                                        \
                     --browser-version "$BROWSER_DRIVER_INSTALLER_FIREFOX_VERSION" \
-                    --target-path "$BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH" || return_code=$?
-else 
-    echo "The environment variables for Firefox are not set, skipping post-install for Firefox"
+                    --target-path "$BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH"
+else
+    echo "The environment variables for Firefox (\$BROWSER_DRIVER_INSTALLER_FIREFOX_VERSION, \$BROWSER_DRIVER_INSTALLER_GECKODRIVER_PATH) are not set, skipping the post-install step for Firefox"
```
Comment (thehcma) [inline postinstall]: same

File: check-environment-and-install [hunk 8]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+[ "${BASH_SOURCE[0]}" != "$0" ] && { echo "ERROR: this script must be executed by Bash, neither source'd nor executed by a different shell"; return 1; }
+
+set -e
+if [[ -n "${BROWSER_DRIVER_INSTALLER_CHROME_VERSION:-}" && -n "${BROWSER_DRIVER_INSTALLER_CHROMEDRIVER_PATH:-}" ]]; then
```
Comment (thehcma) [inline check-environment-and-install:6]: add a blank line after set -e

we probably want set -u as well (undefined vars)

Comment (mzyil) [reply check-environment-and-install:6]: added blank line, adding `set -u` would render the error messages useless

File: postinstall (hunk 7 above)
Comment (mzyil) [reply postinstall]: fixed

File: postinstall (hunk 6 above)
Comment (mzyil) [reply postinstall]: fixed

File: check-environment-and-install (hunk 8 above)
Comment (thehcma) [reply check-environment-and-install:6]: how so?

**Review by thehcma — APPROVED** [review]

**Pull request merged**